sdks/core/src/main/java/org/gbif/pipelines/core/interpreters/core/TemporalInterpreter.java

```diff
@@ -41,80 +48,276 @@
-          .map(LocalDateTime::toString)
-          .ifPresent(x -> tr.setEventDate(new EventDate(x, null)));
-
-      tr.setYear(atomizedLocalDate.getYear());
-      tr.setMonth(atomizedLocalDate.getMonth());
-      tr.setDay(atomizedLocalDate.getDay());
+  /**
+   * Interpreter works on an independent VM, changing a static TemporalParser should be safe.
+   *
+   * <p>Set a temporal parser based on config. Support D/M/Y EU/AU or M/D/Y us formats for ambiguous
+   * dates, like 01/02/2001
+   *
+   * @param config
+   */
+  public static void setTemporalParser(PipelinesConfig config) {
+    if (Strings.isNullOrEmpty(config.getDefaultDateFormat())) {
+      temporalParser = DateParsers.defaultTemporalParser();
+    } else {
+      if (config.getDefaultDateFormat().equalsIgnoreCase("DMY")) {
+        temporalParser = CustomizedTextDateParser.getInstance(DMY_FORMATS);
+      } else if (config.getDefaultDateFormat().equalsIgnoreCase("MDY")) {
+        temporalParser = CustomizedTextDateParser.getInstance(MDY_FORMATS);
+      } else {
+        temporalParser = DateParsers.defaultTemporalParser();
+      }
     }
-    addIssueSet(tr, eventResult.getIssues());
+  }
 
-    LocalDate upperBound = LocalDate.now().plusDays(1);
+  /**
+   * Preprocess for converting some none ISO standards to ISO standards
+   *
+   * @param dateString
+   * @return
+   */
+  private static String normalizeDateString(String dateString) {
+    // Convert 2004-2-1 to 3-2 , 2004-2-1 & 3-2  to 2004-2-1/3-2
+    if (StringUtils.isNotEmpty(dateString)) {
+      dateString = dateString.replace(" to ", "/");
+      dateString = dateString.replace(" & ", "/");
+    }
+
+    return dateString;
+  }
+
+  public static void interpretTemporal(ExtendedRecord er, TemporalRecord tr) {
+    interpretEventDate(er, tr);
+    //    OccurrenceParseResult<TemporalAccessor> eventResult = interpretRecordedDate(er);
+    //    if (eventResult.isSuccessful()) {
+    //      Optional<TemporalAccessor> temporalAccessor =
+    // Optional.ofNullable(eventResult.getPayload());
+    //
+    //      Optional<TemporalAccessor> localDate =
+    //          temporalAccessor
+    //              .filter(ta -> ta.isSupported(ChronoField.HOUR_OF_DAY))
+    //              .map(ta -> ta.query(TemporalQueries.localDate()));
+    //
+    //      if (localDate.isPresent()) {
+    //        temporalAccessor = localDate;
+    //      }
+    //
+    //      temporalAccessor
+    //          .map(TemporalAccessor::toString)
+    //          .ifPresent(x -> tr.setEventDate(new EventDate(x, null)));
+    //
+    //      temporalAccessor
+    //          .map(AtomizedLocalDate::fromTemporalAccessor)
+    //          .ifPresent(
+    //              ald -> {
+    //                tr.setYear(ald.getYear());
+    //                tr.setMonth(ald.getMonth());
+    //                tr.setDay(ald.getDay());
+    //              });
+    //    }
+    //    addIssueSet(tr, eventResult.getIssues());
+  }
```

Comment:
Please, don't leave commented code, just delete, if we need it, we can restore using git history

Reply:
@muttcg  Thanks for your advice. I forgot to remove the comments.  

If there are only three methods should be in, then interpretTemporal should be called interpretEventDate, I think

I have been considering a general range parser,  I found it could be hard, because: for startDate, we need provided year, month, day to solve ambiguous date, mainly on like 01/02/2004.
but for end date, we don't need to that.

Like what you mentioned, passing year, month, day into the method, looks not very generic. 

ParsedTemporal pt = parseDateRange(String year, String month, String day, String dateString);

I am still working on it